OpenRA.Mods.Common/Traits/AutoTarget.cs

```diff
@@ -170,7 +171,9 @@ Actor ChooseTarget(Actor self, WDist range, bool allowMove)
 				// criterion
 				.Select(a => new KeyValuePair<Armament, Actor>(
 					attack.ChooseArmamentsForTarget(Target.FromActor(a), false)
-						.Where(arm => allowMove || arm.MaxRange().Length > (a.CenterPosition - self.CenterPosition).Length)
+						.Where(arm => allowMove
+							|| (Target.FromActor(a).IsInRange(self.CenterPosition, arm.MaxRange())
+							&& !Target.FromActor(a).IsInRange(self.CenterPosition, arm.Weapon.MinRange)))
```

Comment:
Can you pull out a target variable instead of having three `Target.FromActor(a)`?